numpy/__init__.py

```diff
@@ -382,12 +382,13 @@ def _mac_os_check():
             if len(w) > 0:
                 error_message = "{}: {}".format(w[-1].category.__name__, str(w[-1].message))
```

Comment:
This feels quite odd to me - it might just be better to do `warnings.simplefilter('error')` within this `with` block, and then use
```
try:
   _max_os_check()
except BaseWarning as e:  # or whatever this type is called:
   raise RuntimeError("Polyfit sanity test ...") from e
```

Reply:
I don't feel confident in changing it without the ability to test locally, but feel free to push any suggestions up!

Reply:
you can do a local edit to force execution of that code path (without committing the change) by making `if ...` into `if 1 or ....`

Reply:
Yes, but the `_mac_os_check` is platform-specific I believe, so I'm not confident I can meet all of the triggering conditions locally. Either way, the suggestion was outside of the scope of what I had originally intended anyways - if we want to change it here that's fine with me but I leave it to others.